src/main/java/org/springframework/data/elasticsearch/core/indexmapping/MappingBuilder.java

```diff
@@ -106,7 +106,7 @@ public MappingBuilder(ElasticsearchConverter elasticsearchConverter) {
 	 */
 	public String buildPropertyMapping(Class<?> clazz) throws IOException {
 
-		ElasticsearchPersistentEntity<?> entity = elasticsearchConverter.getMappingContext()
+		ElasticsearchPersistentEntity<?> entity = elasticsearchEntityConverter.getMappingContext()
```

Comment:
Do we already have ticket to change `MappingBuilder`'s return type to `Document`? That way, we can remove Jackson usage here entirely and use `PutMappingRequestBuilder setSource(Map mappingSource)`.

Reply:
`buildPropertyMapping` does not use Jackson, it uses Elasticsearch's `XContentBuilder`, it is used in the `addDynamicTemplatesMapping` method where external json files are parsed. Not sure if we can get rid of the `ObjectMapper` here